src/Data/Tree/AVL/Indexed/Relation/Unary/Any/Properties.agda

```diff
@@ -148,15 +163,89 @@ joinʳ⁺-right⁺ k₂ t₁ (1# , node⁻ k₆ k₄ t₃ t₅ bal t₇) ∼+  (
--- insert
+joinˡ⁺⁻ : ∀ {l u hˡ hʳ h} →
+            (kv@(k , v) : K& V) →
+            (l : ∃ λ i → Tree V l [ k ] (i ⊕ hˡ)) →
+            (r : Tree V [ k ] u hʳ) →
+            (bal : hˡ ∼ hʳ ⊔ h) →
+            Any P (proj₂ (joinˡ⁺ kv l r bal)) →
+            P kv ⊎ Any P (proj₂ l) ⊎ Any P r
+joinˡ⁺⁻ k₂ (0# , t₁)                       t₃ ba = Any.toSum
+joinˡ⁺⁻ k₂ (1# , t₁)                       t₃ ∼0 = Any.toSum
+joinˡ⁺⁻ k₂ (1# , t₁)                       t₃ ∼+ = Any.toSum
+joinˡ⁺⁻ k₄ (1# , node k₂ t₁ t₃ ∼-)         t₅ ∼- = λ where
+  (left p)          → inj₂ (inj₁ (left p))
+  (here p)          → inj₂ (inj₁ (here p))
+  (right (left p))  → inj₂ (inj₁ (right p))
+  (right (here p))  → inj₁ p
+  (right (right p)) → inj₂ (inj₂ p)
+joinˡ⁺⁻ k₄ (1# , node k₂ t₁ t₃ ∼0)         t₅ ∼- = λ where
+  (left p)          → inj₂ (inj₁ (left p))
+  (here p)          → inj₂ (inj₁ (here p))
+  (right (left p))  → inj₂ (inj₁ (right p))
+  (right (here p))  → inj₁ p
+  (right (right p)) → inj₂ (inj₂ p)
+joinˡ⁺⁻ k₆ (1# , node⁺ k₂ t₁ k₄ t₃ t₅ bal) t₇ ∼- = λ where
+  (left (left p))   → inj₂ (inj₁ (left p))
+  (left (here p))   → inj₂ (inj₁ (here p))
+  (left (right p))  → inj₂ (inj₁ (right (left p)))
+  (here p)          → inj₂ (inj₁ (right (here p)))
+  (right (left p))  → inj₂ (inj₁ (right (right p)))
+  (right (here p))  → inj₁ p
+  (right (right p)) → inj₂ (inj₂ p)
+
+joinʳ⁺⁻ : ∀ {l u hˡ hʳ h} →
+            (kv : K& V) →
+            (l : Tree V l [ kv .key ] hˡ) →
+            (r : ∃ λ i → Tree V [ kv .key ] u (i ⊕ hʳ)) →
+            (bal : hˡ ∼ hʳ ⊔ h) →
+            Any P (proj₂ (joinʳ⁺ kv l r bal)) →
+            P kv ⊎ Any P l ⊎ Any P (proj₂ r)
+joinʳ⁺⁻ k₂ t₁ (0# , t₃)                       bal = Any.toSum
+joinʳ⁺⁻ k₂ t₁ (1# , t₃)                       ∼0  = Any.toSum
+joinʳ⁺⁻ k₂ t₁ (1# , t₃)                       ∼-  = Any.toSum
+joinʳ⁺⁻ k₂ t₁ (1# , node k₄ t₃ t₅ ∼+)         ∼+  = λ where
+  (left (left p))  → inj₂ (inj₁ p)
+  (left (here p))  → inj₁ p
+  (left (right p)) → inj₂ (inj₂ (left p))
+  (here p)         → inj₂ (inj₂ (here p))
+  (right p)        → inj₂ (inj₂ (right p))
+joinʳ⁺⁻ k₂ t₁ (1# , node k₄ t₃ t₅ ∼0)         ∼+  = λ where
+  (left (left p))  → inj₂ (inj₁ p)
+  (left (here p))  → inj₁ p
+  (left (right p)) → inj₂ (inj₂ (left p))
+  (here p)         → inj₂ (inj₂ (here p))
+  (right p)        → inj₂ (inj₂ (right p))
+joinʳ⁺⁻ k₂ t₁ (1# , node⁻ k₆ k₄ t₃ t₅ bal t₇) ∼+  = λ where
+  (left (left p))   → inj₂ (inj₁ p)
+  (left (here p))   → inj₁ p
+  (left (right p))  → inj₂ (inj₂ (left (left p)))
+  (here p)          → inj₂ (inj₂ (left (here p)))
+  (right (left p))  → inj₂ (inj₂ (left (right p)))
+  (right (here p))  → inj₂ (inj₂ (here p))
+  (right (right p)) → inj₂ (inj₂ (right p))
 
 module _ {V : Value v} where
 
   private
     Val  = Value.family V
     Val≈ = Value.respects V
 
+  singleton⁺ : {P : Pred (K& V) p} →
+               (k : Key) →
+               (v : Val k) → {l<k<u : l < k < u} →
```

Comment:
Should `l<k<u` really be implicit? Can it be reliably reconstructed by Agda from `Any P (singleton k v l<k<u)` in general? Same for the function below.